packages/next/src/server/base-server.ts

```diff
@@ -2373,10 +2373,12 @@ export default abstract class Server<ServerOptions extends Options = Options> {
 
       // Add any fetch tags that were on the page to the response headers.
       const cacheTags = metadata.fetchTags
+
+      // Copy the headers from the response.
+      headers = { ...res.getHeaders() }
```

Comment:
This is the case where we lazily on-demand render a static page or the static part of a PPR page. This one is a bit sketchy. I'm not sure if this is the best way and if this `res` is actually always going to be the same response as what we expect here.

I'm trying to read back the headers we've already added in the render but I'm not sure this is 100% safe.

Reply:
This is the cause of the test failure because this response will have too many headers that we're not actually interested in writing to the meta data in it.

Reply:
Went with another approach: https://github.com/vercel/next.js/pull/58162/commits/f501fce0917aa83923a9c9aa3fbd91f1c317c3d9